### DISCUSSION TITLE
Error : Call to a member function namespace() on null

### DISCUSSION THREAD

**Discussion by Nathan269** [statamic/cms#8485] [Troubleshooting]
This is just the first part of my test that is failing:

    /** @test */
        public function hidden_page_is_removed_from_topics_indexes()
    {
        $entry = Entry::whereCollection('latest_analysis')->first();
        $entry->set('hide_page', false);
        $entry->save();
    }

I get this error:

Error : Call to a member function namespace() on null
 /Users/nathanbayne/projects/vtm3/vendor/withcandour/aardvark-seo/src/Listeners/AppendEntrySeoFieldsListener.php:24

It's complaining about this line within the AppendEntrySeoFieldsListener.php:

$handle = $event->blueprint->namespace();

if i dd($handle) and then run the test, i get this:

collections.pages

does anyone have any idea why namespace is returning as null?

**Comment by ryanmitchell** [comment]
Is it possible some entries have a blueprint and some don't?

**Reply by Nathan269** [reply]
All entries have a set blueprint of "Article" based on what I have in my database seeder. So every entry should have a blueprint set.

**Reply by duncanmcclean** [reply]
What does your seeder look like?

**Reply by Nathan269** [reply]
this is my seeder:

public function run()
    {
        $slug = 'dats-gfai-igms-nlsn-sxtc-trending-stocks-today-29-march';

        Collection::make('latest_analysis')
            ->title('Latest Analysis')
            ->dated(true)
            ->save();

        $entry = Entry::make()
            ->collection('latest_analysis')
            ->slug($slug)
            ->date(now())
            ->data(json_decode('{"bard": [{"type": "paragraph", "content": [{"text": "As Russia-Ukraine talks look promising, US stock markets appear set to open in the green. Most US equities finished higher on Monday, with growth outperforming value.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "Oil", "type": "text", "marks": [{"type": "link", "attrs": {"rel": null, "href": "https://www.valuethemarkets.com/topics/oil-gas-consumable-fuels", "title": null, "target": null}}, {"type": "bold"}, {"type": "underline"}]}, {"text": " down, ", "type": "text"}, {"text": "gold", "type": "text", "marks": [{"type": "link", "attrs": {"rel": null, "href": "https://www.valuethemarkets.com/topics/gold", "title": null, "target": null}}, {"type": "bold"}, {"type": "underline"}]}, {"text": " down and ", "type": "text"}, {"text": "Bitcoin (BTC)", "type": "text", "marks": [{"type": "link", "attrs": {"rel": null, "href": "https://www.valuethemarkets.com/education/what-is-bitcoin-mining", "title": null, "target": null}}, {"type": "bold"}, {"type": "underline"}]}, {"text": " up.", "type": "text"}]}, {"type": "paragraph", "content": [{"type": "hard_break"}, {"text": "Here are some of today\'s trending stocks:", "type": "text"}]}, {"type": "heading", "attrs": {"level": 2}, "content": [{"text": "DatChat Inc (NASDAQ: DATS)", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "DatChat", "type": "text", "marks": [{"type": "bold"}]}, {"text": " (NASDAQ: DATS) is a blockchain, cybersecurity, and social media company. Shares of DatChat Inc (DATS) rose by 61.45% yesterday. Over the past year, DatChat Inc (DATS) has traded between $1.39 and $18.50.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "The company focuses on privacy protection by also protecting messages and posts after they\'ve been sent. It also operates a private, encrypted social network.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "The reason for the increased investor interest in DATS stock yesterday was a company announced it\'s launching a metaverse advertising and NFT monetization platform.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "VenVūū is an advertising platform that allows metaverse landowners and brands to connect using DatChat\'s technology. The platform provides metaverse landowners an easy way to monetize their holdings by opting into advertisers\' paid ad campaigns. Brands can run campaigns across the NFT ad network targeting the land parcels they want to reach across multiple metaverses.", "type": "text"}]}, {"type": "heading", "attrs": {"level": 2}, "content": [{"text": "Guardforce AI Co Ltd (NASDAQ: GFAI)", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "Guardforce AI Co Ltd", "type": "text", "marks": [{"type": "bold"}]}, {"text": " (NASDAQ: GFAI) stock is up 37% in pre-market trading. The company closed up 5% yesterday. The news boosting the GFAI share price is the company plans to expand into Dubai and Australia.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "Guardforce AI is a security company focused on developing robotic solutions and information security services complementing its logistics business. Its services include cash-in-transit, dedicated vehicles to banks, ATM management, cash center operations, cash processing, coin processing, cheque center, and cash deposit machine solutions.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "GFAI stock went public via IPO in September 2021. Year-to-date, the Guardforce AI Co Ltd (GFAI) stock is up by 1.72%, while the S&P 500 is -4.61% over the same period. Meanwhile, the GFAI share price is down 70% since IPO.", "type": "text"}]}, {"type": "heading", "attrs": {"level": 2}, "content": [{"text": "IGM Biosciences Inc (NASDAQ: IGMS)", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "IGM Biosciences Inc", "type": "text", "marks": [{"type": "bold"}]}, {"text": " (NASDAQ: IGMS) stock price is up 39% in pre-market trading on news of a deal with ", "type": "text"}, {"text": "Sanofi", "type": "text", "marks": [{"type": "bold"}]}, {"text": " (NASDAQ: SNY). Sanofi will use IGM\'s antibody tech platform to focus on discoveries surrounding three oncology diseases and three immunology/inflammatory diseases.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "IGM will receive $150m upfront from Sanofi with milestone payments that could amount to nearly $6bn.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "This impressive partnership adds credibility to IGM and makes potential development in revolutionary therapeutics a reality.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "Over the past year, IGM Biosciences Inc (IGMS) has traded between $13.11 and $99.44. It is trading at $20.88 in the pre-market.", "type": "text"}]}, {"type": "heading", "attrs": {"level": 2}, "content": [{"text": "Nielsen Holdings PLC (NYSE: NLSN)", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "Nielsen Holdings PLC", "type": "text", "marks": [{"type": "bold"}]}, {"text": " (NYSE: NLSN), an American TV ratings and marketing data company, is up 21% in pre-market trading.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "The reason for heightened investor interest is a takeover offer of $28 per share.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "Brookfield Business Partners, leading a private equity consortium, raised its buyout offer for Nielsen Holdings to around $16bn after Nielsen rejected a $9bn takeover bid. ", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "Over the past year, Nielsen Holdings PLC (NLSN) has traded between $16.02 and $28.42. It is around $27 today.", "type": "text"}]}, {"type": "heading", "attrs": {"level": 2}, "content": [{"text": "China SXT Pharmaceuticals Inc (NASDAQ: SXTC)", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "China SXT Pharmaceuticals Inc", "type": "text", "marks": [{"type": "bold"}]}, {"text": " (NASDAQ: SXTC) stock is up 43% in pre-market trading. This appears to be a move due to social chatter and high-volume trading. There is no sign of company news.", "type": "text"}]}, {"type": "paragraph", "content": [{"text": "Over the past year, China SXT Pharmaceuticals Inc (SXTC) has traded between $0.11 and $2.28. Today it trades at $0.29. SXTC stock is down 70% year-to-date, so holders will appreciate the rally.", "type": "text"}]}], "funds": [], "title": "DATS, GFAI, IGMS, NLSN, SXTC: Trending Stocks Today 29 March", "main_1": [], "main_2": [], "parent": null, "topics": ["trending-stocks"], "authors": [8], "company": [8536, 6274, 2893], "excerpt": "Trending stocks on Tuesday include DatChat (DATS), Guardforce AI (GFAI), IGM Biosciences (IGMS), Nielsen Holdings (NLSN), and China SXT Pharma (SXTC).", "indices": [], "sidebar": null, "tickers": [8537, 6275, 2894], "og_image": "seo/Cybersecurity-Stock-min.jpeg", "og_title": "DATS, GFAI, IGMS, NLSN, SXTC: Trending Stocks Today 29 March", "exchanges": ["nasdaq", "nyse"], "blueprint": "Article", "hide_page": false, "page_type": "Analysis Article", "sidebar_1": [], "sidebar_2": [], "sidebar_3": [], "full_width": null, "industries": [40, 45, 44], "media_type": "rich_text", "meta_title": "DATS, GFAI, IGMS, NLSN, SXTC: Trending Stocks Today 29 March", "external_id": null, "share_links": true, "meta_section": null, "urls_section": null, "canonical_url": null, "external_news": false, "head_snippets": null, "meta_keywords": [], "no_index_page": false, "twitter_title": null, "featured_image": "Cybersecurity-Stock-min.jpeg", "localized_urls": [], "og_description": "Trending stocks on Tuesday include DatChat (DATS), Guardforce AI (GFAI), IGM Biosciences (IGMS), Nielsen Holdings (NLSN), and China SXT Pharma (SXTC)", "schema_objects": null, "author_was_paid": "No", "client_campaign": null, "footer_snippets": null, "is_article_page": true, "no_follow_links": false, "scripts_section": null, "featured_article": false, "indexing_section": null, "meta_description": "Trending stocks on Tuesday include DatChat (DATS), Guardforce AI (GFAI), IGM Biosciences (IGMS), Nielsen Holdings (NLSN), and China SXT Pharma (SXTC)", "share_section_og": null, "sitemap_priority": "0.5", "crypto_currencies": [], "main_comp_section": null, "use_meta_keywords": false, "sitemap_changefreq": "daily", "twitter_description": null, "additional_datalayer": [], "client_campaign_code": null, "fullwidth_components": [], "global_gtm_identifier": null, "google_search_preview": null, "share_section_twitter": null, "twitter_summary_image": null, "featured_image_caption": "DATS Trending Stocks Today 29 March", "original_publisher_url": null, "twitter_card_type_page": "summary", "author_holds_a_position": "No", "client_campaign_page_no": null, "client_campaign_warning": null, "digitonic_ltd_were_paid": "No", "is_client_campaign_page": false, "original_publisher_name": null, "other_author_disclosure": null, "common_campaign_tracking": [], "other_general_disclosure": null, "is_client_campaign_report": false, "override_twitter_settings": false, "twitter_summary_large_image": null, "digitonic_ltd_hold_a_position": "No", "override_twitter_card_settings": false, "have_company_ies_mentioned_contributed_to_the_article": false}', true));

        $entryModel = $entry->toModel();
        $entryModel->uri = '/analysis/'.$slug;
        $entryModel->save();
    }

**Reply by Nathan269** [reply]
I just realised I attached the wrong data in the above post. I've now updated this to include the blueprint that should be included.
